src/relations/openfga.py

```diff
@@ -114,24 +124,27 @@ def _on_openfga_store_created(self, event: OpenFGAStoreCreateEvent):
             logger.info(f"{event.relation.name} revoked, no store id")
             return
 
-        if event.token_secret_id:
-            secret = self.charm.model.get_secret(id=event.token_secret_id)
-            secret_content = secret.get_content()
-            token = secret_content["token"]
-        if event.token:
-            token = event.token
+        info = self.charm.openfga.get_store_info()
+        if not info:
+            logger.info(f"{event.relation.name} revoked, no store info found")
+            return
+
+        url_components = urlsplit(info.http_api_url)
+        scheme = url_components.scheme
+        address = url_components.hostname
+        http_port = url_components.port
 
         if self.charm.unit.is_leader():
             self.charm._state.openfga = {
-                "store_id": event.store_id,
-                "token": token,
-                "address": event.address,
-                "port": event.port,
-                "scheme": event.scheme,
+                "store_id": info.store_id,
+                "token": info.token,
+                "address": address,
+                "port": http_port,
+                "scheme": scheme,
                 "auth_model_id": None,
             }
 
-        self.charm._update(event)
+            self.charm._update(event)
```

Comment:
Extra indentation here?

Reply:
This is intentional, as the `_update` method will be called for the other units when the `_on_peer_relation_changed` hook is triggered.